content/writing-packages.md

```diff
@@ -0,0 +1,282 @@
+
+The practice of writing NPM packages is [well documented](https://docs.npmjs.com/getting-started/creating-node-modules) and we won't cover it here.
+
+However, if your package depends on an Atmosphere package (which includes the Meteor core packages _as of this writing_), or needs to take advantage of Meteor's [build system](#build-packages), then you may need to write an Atmosphere package.
+
+This article will cover some tips on how to do that.
+
+<h2 id="creating">Creating an Atmosphere Package</h2>
+
+To get started writing a package, use the Meteor command line tool:
+
+```bash
+meteor create --package my-package
+```
+
+If you run this inside an app, it will place the newly generated package in that app's `packages/` directory. Outside an app, it will just create a standalone package directory. The command also generates some boilerplate files for you:
+
+```
+my-package
+├── README.md
+├── package.js
+├── my-package-tests.js
+└── my-package.js
+```
+
+The `package.js` file is the main file in every Meteor package. This is a JavaScript file that defines the metadata, files loaded, architectures, NPM packages, and Cordova packages for your Meteor package.
+
+In this guide article, we will go over some important points for building packages, but we won't explain every part of the `package.js` API. To learn about all of the options, [read about the `package.js` API in the Meteor docs.](http://docs.meteor.com/#/full/packagejs)
+
+<h2 id="adding-files">Adding files and assets</h2>
+
+The main function of an Atmosphere package is to contain source code (JS, CSS, and any transpiled languages) and assets (images, fonts, and more). They act as a unit of code organization, give the developer complete control over which files are loaded where, and can be shared between different applications. 
+
+
+<h3 id="adding-javascript">Adding JavaScript</h3>
+
+To add JavaScript files to a package, specify an entrypoint with [`api.mainModule()`](http://docs.meteor.com/#/full/pack_mainModule) in the package's `onUse` block (this will already have been done by `meteor create --package` above):
+
+
+```js
+Package.onUse(function(api) {
+  api.mainModule('my-package.js');
+});
+```
+
+From that entrypoint, you can `import` other files within your package, [just as you would in an application](structure.html).
+
+If you want to include different files on the client and server, you can specify multiple entry points using the second argument to the function:
+
+```js
+Package.onUse(function(api) {
+  api.mainModule('my-package-client.js', 'client');
+  api.mainModule('my-package-server.js', 'server');
+});
+```
+
+You can also add any source file that would be compiled to a JS file (such as a CoffeeScript file) in a similar way, assuming you [depend](#dependencies) on an appropriate build plugin.
+
+<h3 id="adding-css">Adding CSS</h3>
```

Comment:
I'm not quite sure what he mean here, I can't make it work, will leave this for now, work on for 1.3.1